server/src/main/java/org/elasticsearch/index/mapper/TextFieldMapper.java

```diff
@@ -302,56 +298,42 @@ public Builder(
             IndexMode indexMode,
             IndexAnalyzers indexAnalyzers,
             boolean isSyntheticSourceEnabled,
-            boolean withinMultiField
+            boolean isWithinMultiField
         ) {
-            super(name);
+            super(name, indexCreatedVersion, isSyntheticSourceEnabled, isWithinMultiField);
 
-            this.indexCreatedVersion = indexCreatedVersion;
             this.indexMode = indexMode;
-            this.isSyntheticSourceEnabled = isSyntheticSourceEnabled;
-            this.withinMultiField = withinMultiField;
-
-            this.norms = Parameter.normsParam(m -> ((TextFieldMapper) m).norms, this::normsDefault);
-
-            // If synthetic source is used we need to either store this field
-            // to recreate the source or use keyword multi-fields for that.
-            // So if there are no suitable multi-fields we will default to
-            // storing the field without requiring users to explicitly set 'store'.
-            //
-            // If 'store' parameter was explicitly provided we'll reject the request.
-            // Note that if current builder is a multi field, then we don't need to store, given that responsibility lies with parent field
-            this.store = Parameter.storeParam(m -> ((TextFieldMapper) m).store, () -> {
-                if (multiFieldsNotStoredByDefaultIndexVersionCheck(indexCreatedVersion)) {
-                    return isSyntheticSourceEnabled
-                        && this.withinMultiField == false
-                        && multiFieldsBuilder.hasSyntheticSourceCompatibleKeywordField() == false;
-                } else {
-                    return isSyntheticSourceEnabled && multiFieldsBuilder.hasSyntheticSourceCompatibleKeywordField() == false;
-                }
-            });
             this.analyzers = new TextParams.Analyzers(
                 indexAnalyzers,
                 m -> ((TextFieldMapper) m).indexAnalyzer,
                 m -> (((TextFieldMapper) m).positionIncrementGap),
                 indexCreatedVersion
             );
-        }
 
-        private boolean normsDefault() {
-            if (indexCreatedVersion.onOrAfter(IndexVersions.DISABLE_NORMS_BY_DEFAULT_FOR_LOGSDB_AND_TSDB)) {
-                // don't enable norms by default if the index is LOGSDB or TSDB based
-                return indexMode != IndexMode.LOGSDB && indexMode != IndexMode.TIME_SERIES;
-            }
-            // bwc - historically, norms were enabled by default on text fields regardless of which index mode was used
-            return true;
-        }
```

Comment:
Calling this outside of the constructor was bad practice, hence I moved it into the constructor. See the 313-320 below